src/main/3.2/scala/com/crealytics/spark/v2/excel/ExcelTable.scala

```diff
@@ -74,36 +74,48 @@ case class ExcelTable(
       sample = if (sample < 1) 1 else sample
       inputPaths.take(sample).map(_.getPath.toUri)
     }
-    var rows = excelHelper.getRows(conf, paths.head)
+    var sheetData = excelHelper.getSheetData(conf, paths.head)
 
-    if (rows.isEmpty) { /* If the first file is empty, not checking further */
+    if (sheetData.rowIterator.isEmpty) { /* If the first file is empty, not checking further */
       StructType(Seq.empty)
     } else {
-      /* Prepare field names */
-      val colNames =
-        if (options.header) { /* Get column name from the first row */
-          val r = excelHelper.getColumnNames(rows.next())
-          rows = rows.drop(options.ignoreAfterHeader)
-          r
-        } else { /* Peek first row, then return back */
-          val headerRow = rows.next()
-          val r = excelHelper.getColumnNames(headerRow)
-          rows = Iterator(headerRow) ++ rows
-          r
+      try {
```

Comment:
Same here. I had not seen that there is so much overlap in the code, but we should get rid of it.

Reply:
I've done some refactoring to reduce code duplication - probably more can be done